projects/canopy/src/lib/alert/alert.component.ts

```diff
@@ -30,20 +30,30 @@ export class LgAlertComponent {
 
     this.renderer.addClass(this.hostElement.nativeElement, `lg-variant--${variant}`);
     this._variant = variant;
+    this.setRole();
   }
   get variant() {
     return this._variant;
   }
 
   @HostBinding('class.lg-alert') class = true;
+  @HostBinding('attr.role') _role: string;
 
-  @HostBinding('attr.role') get role(): string {
-    if (this.variant !== 'info' && this.variant !== 'generic') {
-      return 'alert';
-    }
+  constructor(
+    private renderer: Renderer2,
+    private hostElement: ElementRef,
+  ) {
+    this.variant = 'generic';
   }
 
-  constructor(private renderer: Renderer2, private hostElement: ElementRef) {
-    this.variant = 'generic';
+  @Input() set role(role: string) {
+    this._role = role;
+  }
+  private setRole() {
+    if (!this._role) {
```

Comment:
Could we combine the nested `if` conditions? I can't see any value to separating them in this case.

Reply:
Sure thing!

Reply:
I'm not sure I completely agree. I see this `if` as being a sort of 'guard'. I think it's possibly easier to read too, I'm not keen on `if` statements containing multiple logical operators (especially when dealing with Not).

Reply:
if we're going to keep it separate to escape early, would it be better to make it explicit and have something along the lines of...
```
private setRole() { 
  if (!this._role) return

  if...```

Reply:
https://github.com/Legal-and-General/canopy/blob/c2d78bfb37bbc4ccfa0a495b341a253f2deee2e5/projects/canopy/src/lib/accordion/accordion-item/accordion-item.component.ts#L117

All roads lead to Rome. I merged the if clauses as per the initial request.

Reply:
@pauleustice I also added role input in storybook.